channeldb/forwarding_log_test.go

```diff
@@ -360,3 +361,213 @@ func TestForwardingLogStoreEvent(t *testing.T) {
 		}
 	}
 }
+
+// TestForwardingLogQueryIncomingChanIDs tests that querying the forwarding log
+// using only incoming channel IDs returns the correct subset of events.
+func TestForwardingLogQueryIncomingChanIDs(t *testing.T) {
+	t.Parallel()
+
+	// Set up a test database.
+	db, err := MakeTestDB(t)
+	require.NoError(t, err, "unable to make test db")
+
+	log := ForwardingLog{
+		db: db,
+	}
+
+	initialTime := time.Unix(1234, 0)
+	endTime := time.Unix(1234, 0)
+
+	// Create 10 random events with varying incoming ChanIDs.
+	numEvents := 10
+	events := make([]ForwardingEvent, numEvents)
+	incomingChanIDs := []lnwire.ShortChannelID{
+		lnwire.NewShortChanIDFromInt(2001),
+		lnwire.NewShortChanIDFromInt(2002),
+		lnwire.NewShortChanIDFromInt(2003),
+	}
+
+	for i := 0; i < numEvents; i++ {
+		events[i] = ForwardingEvent{
+			Timestamp:      endTime,
+			IncomingChanID: incomingChanIDs[i%len(incomingChanIDs)],
+			AmtIn:          lnwire.MilliSatoshi(rand.Int63()),
+			AmtOut:         lnwire.MilliSatoshi(rand.Int63()),
+		}
+		endTime = endTime.Add(time.Minute * 10)
+	}
+
+	// Add events to the database.
+	require.NoError(t, log.AddForwardingEvents(events),
+		"unable to add events")
+
+	// Query with multiple incoming channel IDs.
+	eventQuery := ForwardingEventQuery{
+		StartTime: initialTime,
+		EndTime:   endTime,
+		IncomingChanIDs: fn.NewSet(incomingChanIDs[0].ToUint64(),
+			incomingChanIDs[1].ToUint64()),
+		IndexOffset:  0,
+		NumMaxEvents: 10,
+	}
+	timeSlice, err := log.Query(eventQuery)
+	require.NoError(t, err, "unable to query for events")
+
+	// Verify that only events with the specified incomingChanIDs are
+	// returned.
+	expectedEvents := []ForwardingEvent{}
+	for _, e := range events {
+		if e.IncomingChanID == incomingChanIDs[0] ||
+			e.IncomingChanID == incomingChanIDs[1] {
+
+			expectedEvents = append(expectedEvents, e)
+		}
+	}
+
+	require.Equal(t, expectedEvents, timeSlice.ForwardingEvents,
```

Comment:
See my comment above. IMO those additional context strings can just be omitted to make things tidier. The generated error message if this fails is pretty clear where things fail and what expectation failed. So often no additional context is needed in the first place.